### ISSUE TITLE
[Feature Request] 7z archive format support on Linux builds

### ISSUE THREAD

**Issue by it9exm** [TASEmulators/fceux#74]
In the windows builds, I am able to open 7z archives and select the rom or fds image from a list. Why can't we do this on the linux builds too?

**Comment by zeromus** [comment]
because the linux port is on life support and is entirely separate frontend code than windows. Do not expect this to ever happen. However I'll leave the thread open and start linking other "why not in linux" feature requests to it so that all the things that will never happen can stay in here

**Comment by axredneck** [comment]
Meanwhile FCEUX works good under Wine. Maybe support Wine officially?

**Comment by keithbowes** [comment]
> because the linux port is on life support

That's a shame.  That must be why my SDL2 PR was rejected.  Oh well, I now use RetroArch (with the fceumm core), which has the same Wayland and KMS/DRM support that SDL2 does (plus 7z support, realtime movie recording, etc).

Eh, if you're not going to care about doing anything for *nix, perhaps it's time to just pull the plug.

**Comment by bbbradsmith** [comment]
> > because the linux port is on life support
> That's a shame. That must be why my SDL2 PR was rejected.

No, your PR #17 was closed because you were asked "why" and never provided a reason after several months. "Eh, it's compilable" is not a reason to merge the PR.

If you had finished working on that branch so that it actually functions as an SDL2 app, rather than just compiling, and maybe responded with an explanation that it has Wayland / KMS/DRM support and why that's important, it might become worth merging.

> Eh, if you're not going to care about doing anything for *nix, perhaps it's time to just pull the plug.

There's no "plug", we just don't currently seem to have active maintainers for these platforms. It's open source, and at any moment someone capable could take up that task, but this is the current situation.

**Comment by mjbudd77** [comment]
I have added to the Qt/SDL port the ability to extract and open ROMs from zip files. This has been achieved by linking the zlib/minizip system libraries into fceux. However, these libraries do not currently support 7zip file formats. I looked around and there doesn't seem to be a good third party library for linux/macOSX that can provide a nice programming interface to embed the functionality into the fceux executable itself. There is a command line utility called 7za that is part of the p7zip project and is supported by both OS's. It would be possible for fceux to accomplish extracting and opening ROMs from a 7zip archive by using the 7za executable via its command line interface. Issuing commands to the executable and interpreting its output as needed. But I question why this is really worth the trouble. If you want to load a ROM from a 7zip archive, extract it with the system tools that you already have available to you and then it will be loadable by fceux. Or just use regular zip instead of 7zip.
